Code/autopkglib/SignToolVerifier.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/local/autopkg/python
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""See docstring for SignToolVerifier class"""
+
+import os.path
+import subprocess
+from typing import Any, Dict, List, Optional
+
+from autopkglib import Processor, ProcessorError
+
+__all__ = ["SignToolVerifier"]
+
+
+class SignToolVerifier(Processor):
+    """Verifies an authenticode signed installer using the Microsoft SDK
+    signtool executable."""
+
+    EXTENSIONS: List[str] = [".exe", ".msi"]
+
+    # TODO: How much of this is needed to act as a drop-in replacement in an
+    # override recipe??
+    input_variables: Dict[str, Any] = {
+        "DISABLE_CODE_SIGNATURE_VERIFICATION": {
+            "required": False,
+            "description": ("Prevents this processor from running."),
+        },
+        "input_path": {
+            "required": True,
+            "description": (
+                "File path to an `.msi` or `.exe` file for Authenticode verification",
+            ),
+        },
+        "signtool_path": {
+            "required": False,
+            "description": (
+                "The path to signtool.exe. This varies between versions of the "
+                "Windows SDK, so you can explicitly set that here in an override."
+            ),
+            "default": r"C:\Program Files (x86)\Windows Kits\10\bin\x64\signtool.exe",
+        },
+        "additional_arguments": {
+            "required": False,
+            "description": (
+                "Array of additional argument strings to pass to signtool. "
+                "Note that currently '/v' and '/pa' are always passed."
+            ),
+            "default": None,
+        },
+    }
+    output_variables: Dict[str, Any] = {}
+
+    description: str = __doc__
+
+    def codesign_verify(
+        self,
+        path: str,
+        additional_arguments: Optional[List[str]] = None,
+        signtool_path: Optional[str] = None,
+    ) -> bool:
+        """
+        Runs 'signtool.exe /pa <path>'. Returns True if signtool exited with 0
+        and False otherwise.
+        """
+        if not additional_arguments:
+            additional_arguments = []
+
+        signtool_bin = signtool_path if signtool_path else "signtool.exe"
```

Comment:
I'd greatly prefer an absolute path here. I know it's not very Windows-like, but relative paths keep me up at night.

Reply:
Oh, thanks for calling this out. This does need to change. What would you think of getting rid of this fallback entirely? I'm thinking that since `signtool_path` has an absolute path as its default value in the input variables, it should be fine to have the only two cases here be: using the default, or manually specifying the variable. In either case, if the binary doesn't exist, the processor should error out explicitly.

Reply:
I'm okay with just eliminating it entirely, if it's defined in the input with a valid default. I made a comment about this specifically - if we're going to provide the path in the default value for the input variable, let's not hardcode C:.